x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1175,6 +1186,20 @@ LogEntryBuilder withRequestBody(PutRoleRequest putRoleRequest) throws IOExceptio
             return this;
         }
 
+        LogEntryBuilder withRequestBody(BulkPutRolesRequest bulkPutRoleRequest) throws IOException {
+            logEntry.with(EVENT_ACTION_FIELD_NAME, "bulk_put_role");
+            XContentBuilder builder = JsonXContent.contentBuilder().humanReadable(true);
+            builder.startObject().startObject("roles");
+
+            for (RoleDescriptor roleDescriptor : bulkPutRoleRequest.getRoles()) {
+                withRoleDescriptor(builder.field(roleDescriptor.getName()), roleDescriptor);
+            }
+
+            builder.endObject().endObject();
+            logEntry.with(CHANGE_CONFIG_FIELD_NAME, Strings.toString(builder));
```

Comment:
This is a good option.
Another option is to create separate audit entries for every role, maintaining the format of the individual audit entry. That's important because audit entries could be indexed.

I suggest hearing what the broader team has to say. Could you please raise it in the team meeting today?

Reply:
Yes, good point. Thanks for raising this.

Reply:
Raised this in the weekly security meeting and we agreed to change this to separate audit entries per role. Thanks for the suggestion @albertzaharovits !